### DISCUSSION TITLE
ISAPI over HPP_OpenApi

### DISCUSSION THREAD

**Discussion by trempetic92** [petrleocompel/hikaxpro_hacs#144] [Ideas]
Testing usage of OpenApi endpoints on hik partner pro sever. And it seems everything works. Panel does not need to be on local network anymore. As long its added to HPP everything works.

All Local ISAPI will work when pushing over HPP OpenApi

Consider if not already, to give user option how panel is added to the system. The nodes remain the same and the body, but url will have prefix: example for ACS i made:

https://hik-partner.com/api/hpcgw/v1/device/transparent/ISAPI/Intelligent/FDLib/FaceDataRecord?format=json

HPP ISAPI tunel: https://hik-partner.com/api/hpcgw/v1/device/transparent/
ISAPI URL: /ISAPI/Intelligent/FDLib/FaceDataRecord?format=json

Header needs to contain X-Devserial and Authorization: Bearer token(generated from Apikey + secret - Hik partner pro)

If you already done this, please close this discussion.

![image](https://github.com/user-attachments/assets/a823180c-6541-4caf-99a6-168e5d99369d)

**Comment by petrleocompel** [comment]
I definitely dont want to convert this integration to "cloud" integration. 

But ISAPI was so far enough to decode all the stuff.
